### ISSUE TITLE
[Bug] Unexpected warning "plugin_parsetree/code_list_on_singleton/list_function_on_singleton"

### ISSUE THREAD

**Issue by david-maison** [OCamlPro/typerex-lint#172]
With the following code
```
type t = A | B | C

let () =
  List.fold_left (fun acc i -> i + acc) 0 [ 0; 2; 3 ];
  List.fold_left (fun _acc i -> i) 0 [ A; B; C ]
```

I got the following warning:
```
File "pouet.ml", line 5, characters 39-40:
  Warning 2: plugin_parsetree/code_list_on_singleton/list_function_on_singleton SEVERITY 7
  List.fold_left is used on a singleton.
```

O_o?
